### PR TITLE
feat(chat): implemented and updated tests

### PR DESCRIPTION
**Pull request by nartovm**
**Description:**

implemented and updated tests

**Checklist:**

- [ ] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [ ] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts [hunk 1]
```diff
@@ -351,8 +351,18 @@ export class ConversationData extends FolderData {
...
+  // public prepareNestedFolder(nestedLevel: number) {
+  //   return super.prepareNestedFolder(nestedLevel, FolderType.Chat);
+  // }
+  public prepareNestedFolder(
+    nestedLevel: number,
+    folderNames?: Record<number, string>,
+  ) {
+    return super.prepareNestedFolder(
+      nestedLevel,
+      FolderType.Prompt,
```
Comment (irinakartun) [inline apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: `FolderType.Chat` should be used here. There is a separate class `promptData.ts` to build prompt folders

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts (hunk 1 above)
Comment (nartovm) [reply apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: Done

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/utils/themesUtil.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { EntityType, Theme } from '@/src/testData';
+import { Colors } from '@/src/ui/domData';
+
+export class ThemesUtil {
+  public static getEntityColors(
```
Comment (irinakartun) [inline apps/chat-e2e/src/utils/themesUtil.ts]: let's use a more descriptive method name - the current name does not clearly indicate that the method returns check-box and entity background colors

File: apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts [hunk 3]
```diff
@@ -101,6 +129,41 @@ export class SideBarEntityAssertion<T extends SideBarEntities> {
       .toBe(expectedColor);
   }
 
+  public async assertEntityCheckboxColor(
+    entity: TreeEntity,
+    expectedColor: string,
+  ) {
+    const entityCheckboxColor = await this.sideBarEntities
+      .getEntityCheckbox(entity.name, entity.index)
+      .evaluate((el) => window.getComputedStyle(el).getPropertyValue('color'));
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts]: please, use already implemented in `BaseElement` method `getComputedStyleProperty(property: string)`.
To use it, you will need to intro a new var in `SideBarEntities` class that refer to `BaseElement`:

```
getEntityCheckboxElement(name: string, index?: number) {
    return this.createElementFromLocator(this.getEntityCheckbox(name, index));
}
```

File: apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts [hunk 4]
```diff
@@ -101,6 +129,41 @@ export class SideBarEntityAssertion<T extends SideBarEntities> {
...
+      .toBe(expectedColor);
+  }
+
+  public async assertEntityCheckboxBorderColors(
+    entity: TreeEntity,
+    expectedColor: string,
+  ) {
+    const entityCheckboxBorderColors = await this.sideBarEntities
+      .getEntityCheckbox(entity.name, entity.index)
+      .evaluate((el) => {
```
Comment (irinakartun) [inline apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts]: please, use already implemented in `BaseElement` method `getAllBorderColors()`

File: apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts [hunk 5]
```diff
@@ -36,55 +36,65 @@ dialTest(
...
     let nestedFolders: FolderInterface[];
     let nestedPrompts: Prompt[] = [];
-    let rootFolder: FolderPrompt;
+    let folderWithPrompts: FolderPrompt;
     let singlePrompt: Prompt;
     let theme: string;
-    let expectedCheckboxColor: string;
-    let expectedEntityBackgroundColor: string;
+    const emptyFolderName = ExpectedConstants.newPromptFolderWithIndexTitle(1);
+    await localStorageManager.setPromptCollapsedSection('Organization');
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts:43]: let's intro enum for collapsed section values

**Review by irinakartun — CHANGES_REQUESTED** [review]

File: apps/chat-e2e/src/utils/themesUtil.ts (hunk 2 above)
Comment (nartovm) [reply apps/chat-e2e/src/utils/themesUtil.ts]: Done

File: apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts (hunk 3 above)
Comment (nartovm) [reply apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts]: done

File: apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts (hunk 4 above)
Comment (nartovm) [reply apps/chat-e2e/src/assertions/sideBarEntityAssertion.ts]: done

File: apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts (hunk 5 above)
Comment (nartovm) [reply apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts:43]: done

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1964.nightly-test.deltixhub.io
>E2E tests status: success

**Review by irinakartun — APPROVED** [review]

**Pull request merged**